DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs

```diff
@@ -55,44 +57,41 @@ public partial class UserDataService
         public List<DelegateUserCard> GetDelegateUserCardsByCentreId(int centreId)
         {
             return connection.Query<DelegateUserCard>(
-                @"SELECT
-                        cd.CandidateID AS Id,
-                        cd.CandidateNumber,
-                        ct.CentreName,
-                        cd.CentreID,
-                        cd.DateRegistered,
-                        ct.Active AS CentreActive,
-                        cd.EmailAddress,
-                        cd.FirstName,
-                        cd.LastName,
-                        cd.Password,
-                        cd.Approved,
-                        cd.Answer1,
-                        cd.Answer2,
-                        cd.Answer3,
-                        cd.Answer4,
-                        cd.Answer5,
-                        cd.Answer6,
-                        cd.JobGroupId,
-                        jg.JobGroupName,
-                        cd.SelfReg,
-                        cd.ExternalReg,
-                        cd.Active,
-                        (SELECT AdminID
-                         FROM AdminUsers au
-                         WHERE (au.Email = cd.EmailAddress
-                                OR au.Email = cd.AliasID)
-                         AND au.Password = cd.Password
-                         AND au.CentreID = cd.CentreID
-                         AND au.Email != ''
-                        ) AS AdminID,
-                        cd.AliasID
+                @$"SELECT {DelegateUserSelectQuery} 
                     FROM Candidates AS cd
                     INNER JOIN Centres AS ct ON ct.CentreID = cd.CentreID
                     INNER JOIN JobGroups AS jg ON jg.JobGroupID = cd.JobGroupID
                     WHERE cd.CentreId = @centreId AND cd.Approved = 1",
                 new { centreId }
             ).ToList();
         }
+
+        
+        public List<DelegateUserCard> GetDelegatesNotRegisteredForGroupByGroupId(int groupId, int centreId)
+        {
+            return connection.Query<DelegateUserCard>(
+                @$"SELECT {DelegateUserSelectQuery} 
+                    FROM Candidates AS cd
+                    INNER JOIN Centres AS ct ON ct.CentreID = cd.CentreID
+                    INNER JOIN JobGroups AS jg ON jg.JobGroupID = cd.JobGroupID
+					LEFT JOIN GroupDelegates gd ON cd.CandidateID = gd.DelegateID AND gd.GroupID = groupId

```

Comment:
Last part should be `gd.GroupID = @groupId` i.e. missing the `@` symbol.

Reply:
Sorry, yes, I added @ to goupId